spec/FrameSpec.js

```diff
@@ -0,0 +1,299 @@
+
+
+  it("should set balls remaining to two on last frame if a strike", function() {
+    lastFrame.play(10);
+    expect(lastFrame._ballsRemaining).toEqual(2);
+  });
+
+  it("should set balls remaining to one on last frame if a spare", function() {
+    lastFrame.play(3);
+    lastFrame.play(7);
+    expect(lastFrame._ballsRemaining).toEqual(1);
+  });
+
+  describe("Last frame not a strike or spare", function() {
+
+    beforeEach(function() {
+      lastFrame = new Frame(true);
+      lastFrame.play(3);
+      lastFrame.play(6);
+    });
+
+    it("shouldn't set bonus balls on the last frame if not a strike or spare", function() {
+      expect(lastFrame._bonusBalls).toEqual(0);
+    });
+
+    it("shouldn't set balls remaining on the last frame if not a strike or spare", function() {
+      expect(lastFrame._ballsRemaining).toEqual(0);
+    });
+
+    it("should set score to 9 on last frame with 3 & 6 pins", function() {
+      expect(lastFrame._score).toEqual(9);
+    });
+
+    it("should set frame complete after two balls on last frame if score <10", function() {
+      expect(lastFrame.isComplete()).toEqual(true);
+    });
+
+  });
+  describe("Ensure correct pins added for each ball", function() {
+
+    beforeEach(function() {
+      frame = new Frame();
+    });
+
+    it("should set ball1 to X after strike", function() {
+      frame.play("X");
+      expect(frame.ball1).toEqual(10);
+    });
+    it("should set ball1 to - after first play", function() {
+      frame.play("-");
+      expect(frame.ball1).toEqual(0);
+    });
+    it("should set ball1 to 7 after first play", function() {
+      frame.play(7);
+      expect(frame.ball1).toEqual(7);
+    });
+    it("should set ball2 to 10 after spare", function() {
+      frame.play("-");
+      frame.play("/");
+      expect(frame.ball2).toEqual(10);
+    });
+    it("should set ball2 to 7 after spare 3&7", function() {
+      frame.play(3);
+      frame.play(7);
+      expect(frame.ball2).toEqual(7);
+    });
+    it("should set ball2 to 6 after second play", function() {
+      frame.play(3);
+      frame.play(6);
+      expect(frame.ball2).toEqual(6);
+    });
+  });
+
+  describe("Pins remaining", function() {
+
+    beforeEach(function() {
+      frame = new Frame();
+    });
+
+    it("should be 10 before play starts", function() {
+      expect(frame.pinsRemaining()).toEqual(10);
+    });
+
+    it("should be 5 if 5 already down", function() {
+      frame.play(5);
+      expect(frame.pinsRemaining()).toEqual(5);
+    });
+
+    it("should be 0 for a spare", function() {
+      frame.play(5);
+      frame.play(5);
+      expect(frame.pinsRemaining()).toEqual(0);
+    });
+
+    it("should be 0 for a strike", function() {
+      frame.play(10);
+      expect(frame.pinsRemaining()).toEqual(0);
+    });
+
+    it("should be 10 remaining on each ball in last frame if 3 strikes", function() {
+      lastFrame = new Frame(true);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
```

Comment:
We have many expectations in the same test - this is a form of duplication. We've already tested elsewhere that pinsRemaining behaves as expected. 

If we were to test that we could roll 3 strikes on the last frame (but no more), I think we'd have confidence that this code was working. I think we do test that further down, so I don't feel like these tests add anything.